### ISSUE TITLE
RepoSense with Morphing

### ISSUE THREAD

**Issue by XenonShawn** [nus-cs2103-AY2223S1/forum#412]
My team decided to morph AB3 rather than evolve it, and I changed a lot of the names, models, test cases, etc. However, because I changed the names of the files/modules, even if I barely touched the contents of the file, the whole file is attributed to me in the RepoSense. Do I need to do anything for this? What about those files which I edited half of it but also renamed it?

**Comment by damithc** [comment]
@XenonShawn if some code is attributed to you incorrectly, you can override that using RepoSense's `@@author` tags mechanism. More details https://nus-cs2103-ay2223s1.github.io/website/admin/tools.html#tool-reposense-for-authorship-tracking

**Comment by XenonShawn** [comment]
> @XenonShawn if some code is attributed to you incorrectly, you can override that using RepoSense's `@@author` tags mechanism. More details https://nus-cs2103-ay2223s1.github.io/website/admin/tools.html#tool-reposense-for-authorship-tracking

Hi Prof, I'm aware of this. I'm just wondering about the extent - am I supposed to `@@author` everything except for the exact lines I changed? That's easily dozens of files, is there a better way than this?

**Comment by damithc** [comment]
> Hi Prof, I'm aware of this. I'm just wondering about the extent - am I supposed to `@@author` everything except for the exact lines I changed? That's easily dozens of files, is there a better way than this?

@XenonShawn Claiming authorship of very small code fragments is allowed, but not recommended. Some more guidance on this aspect (given in the same page):

![image](https://user-images.githubusercontent.com/1673303/199464694-d84e566b-9b42-4ef7-8be0-2cd68bd168cf.png)

**Comment by sltsheryl** [comment]
Could I also check if we should put the `//@@author` comment before or after all the import statements because if we changed the names of the packages, all the import statements would be allocated to us? Or does this not matter since this is just to check the code quality of what we wrote?

**Comment by damithc** [comment]
> Could I also check if we should put the `//@@author` comment before or after all the import statements because if we changed the names of the packages, all the import statements would be allocated to us? Or does this not matter since this is just to check the code quality of what we wrote?

Yes @sltsheryl, the import statements are not that relevant for that purpose. So, no need to bother about the authorship of import statements.
